optuna_integration/_lightgbm_tuner/optimize.py

```diff
@@ -267,8 +275,14 @@ def _postprocess(
             trial._trial_id, _AVERAGE_ITERATION_TIME_KEY, average_iteration_time
         )
         trial.storage.set_trial_system_attr(trial._trial_id, _STEP_NAME_KEY, self.step_name)
+        lgbm_params = copy.deepcopy(self.lgbm_params)
+        custom_objective = _get_custom_objective(lgbm_params)
+        if custom_objective is not None:
+            # NOTE(nabenabe): If custom_objective is not None, custom_objective is not
+            # serializable, so we store its name instead.
+            lgbm_params["objective"] = custom_objective.__name__
```

Comment:
I found out that some objects do not have `__name__`, so I made this part more robust. 
```suggestion
            lgbm_params["objective"] = (
                custom_objective.__name__ if hasattr(custom_objective, ".__name__") else str(custom_objective)
            )
```